libs/nx-terraform/src/generators/create-environment-sp/generator.ts

```diff
@@ -83,20 +101,34 @@ export default async function (
             stdio: 'inherit',
         })
 
-        const { stdout } = await execa(`az`, [
+        const { stdout: stdoutAdList } = await execa(`az`, [
             'ad',
             'sp',
             'list',
             '--display-name',
             servicePrincipalName,
         ])
-        const servicePrincipalObjectId: string = JSON.parse(stdout)[0].id
+        const servicePrincipalObjectId: string = JSON.parse(stdoutAdList)[0].id
         console.log(`Service principal id: ${servicePrincipalObjectId}`)
 
+        const { stdout: stdoutAppList } = await execa(`az`, [
+            'ad',
+            'app',
+            'list',
+            '--display-name',
+            servicePrincipalName,
+        ])
+        const appObjectId: string = JSON.parse(stdoutAppList)[0].id
```

Comment:
It now gathers and outputs the AD Object ID as well as the app id and the app Object ID.

Different commands need one of these ids.

It then stores all 3 in the dev markdown.